### ISSUE TITLE
Suspending CronJob will break CronJobs with startingDeadlineSeconds

### ISSUE THREAD

**Issue by hjacobs** [hjacobs/kube-downscaler#100]
The code sets `startingDeadlineSeconds` to zero:

https://github.com/hjacobs/kube-downscaler/blob/80aee0772613523b45bf271085cc794333335f0f/kube_downscaler/scaler.py#L226

This will break CronJobs. Kubernetes docs: https://kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/#starting-deadline

See also #99

**Comment by hjacobs** [comment]
Related: https://github.com/zalando-incubator/kubernetes-on-aws/pull/3150

**Comment by graillus** [comment]
Why is the `CronJob`'s `spec.startingDeadlineSeconds` reset to 0 in the first place ? I guess there is a valid reason behind this, but why not leaving the original value ?

**Comment by hjacobs** [comment]
@graillus I have the same question.

**Comment by graillus** [comment]
I did some tests on 2 different GKE clusters (kubernetes 1.15):

- On the first cluster, I created 2 CronJobs, scheduled every minute, one with `startingDeadlineSeconds` set to 60, the other to 0.
- On the other cluster, running slighlty modified version of the `kube-downscaler` (without the set `spec.startingDeadlineSeconds` to 0 on resume), same CronJobs.
- Annotated all these CronJobs to make them enter a downtime period for 10 minutes.

Observations after downtime:

- Only CronJob running with patched downscaler and with original `startingDeadlineSeconds` different from 0 resumed as expected.
- Setting the `startingDeadlineSeconds` to 0 seems to give no chance to `CronJobs` to schedule new jobs after resume.

Looks like setting the `startingDeadlineSeconds` to 0 is never a good idea :/
